packages/lab/src/Flow/stories/BaseHook/TeapotNode.tsx

```diff
@@ -0,0 +1,156 @@
+
+import { useHvNode } from "../../hooks/useNode";
+import { isConnected, renderedIcon } from "../../Node/utils";
+
+const classes = {
+  root: css({
+    borderRadius: "25px",
+    backgroundColor: theme.colors.atmo1,
+    boxShadow: theme.colors.shadow,
+    borderWidth: "1px",
+    minWidth: "200px",
+    minHeight: "100px",
+  }),
+  content: css({
+    display: "flex",
+    flexDirection: "row",
+    alignItems: "center",
+    minHeight: "100px",
+    padding: theme.spacing("sm"),
+  }),
+  title: css({
+    color: theme.colors.base_dark,
+  }),
+  cornerIcon: css({
+    position: "absolute",
+    top: `calc(-30px + ${theme.space.sm})`,
+    right: `calc(-22px + ${theme.space.sm})`,
+  }),
+  contentIcon: css({
+    width: 48,
+    height: 48,
+    display: "flex",
+    justifyContent: "center",
+    alignItems: "center",
+    borderRadius: theme.radii.round,
+    marginRight: theme.space.xs,
+  }),
+  nodeToolbar: css({
+    backgroundColor: theme.colors.atmo1,
+    borderRadius: theme.radii.full,
+  }),
+  handle: css({
+    backgroundColor: theme.colors.secondary_80,
+    height: 10,
+    width: 10,
+  }),
+};
+
+const defaultInputs = [
+  {
+    label: "inputs",
+    accepts: ["water", "leafs"],
+  },
+];
+
+const defaultOutputs = [
+  {
+    label: "tea",
+    provides: "tea",
+  },
+];
+
+export const TeapotNode: HvFlowNodeFC<any> = ({
+  id,
+  title: titleProp,
+  subtitle: subtitleProp,
+  groupId = "teapot",
+  color: colorProp,
+  icon: iconProp,
+  inputs: inputsProp = defaultInputs,
+  outputs: outputsProp = defaultOutputs,
+}: HvFlowNodeProps<unknown>) => {
+  const {
+    toggleShowActions,
+    getNodeToolbarProps,
+    handleDefaultAction,
+    nodeActions,
+    title,
+    icon,
+    color,
+    subtitle,
+    showActions,
+    outputEdges,
+  } = useHvNode({
+    id,
+    title: titleProp,
+    subtitle: subtitleProp,
+    color: colorProp,
+    inputs: inputsProp,
+    outputs: outputsProp,
+    icon: iconProp,
+    groupId,
+  });
+  return (
+    <div
+      className={cx(
+        "nowheel",
+        classes.root,
+        css({ borderColor: showActions ? theme.colors.neutral : color }),
+      )}
+      onMouseEnter={toggleShowActions}
+      onMouseLeave={toggleShowActions}
+    >
+      <NodeToolbar className={classes.nodeToolbar} {...getNodeToolbarProps()}>
+        {nodeActions?.map((action) => (
+          <HvIconButton
+            key={action.id}
+            title={action.label}
+            onClick={() => handleDefaultAction(action)}
+          >
+            {renderedIcon(action.icon)}
+          </HvIconButton>
+        ))}
+      </NodeToolbar>
+      <div className={classes.cornerIcon}>
+        {isConnected(id, "source", "0", outputEdges) ? (
+          <Level0Good color="positive" />
+        ) : (
+          <Level2Average color="warning" />
+        )}
+      </div>
+      <Handle
+        className={classes.handle}
+        type="target"
+        position={Position.Left}
+        id="0"
```

Comment:
Currently, the index of the array of inputs or outputs is being used as the id. This makes it so that when I add an id of 1 the checks of the connection won't work. I think we should open a separate issue to tackle this.

Reply:
Sorry, I also got it wrong: one is an output and the other one is an input. Nothing wrong with having the same id 🙏